app/src/src_nonfreenet/org/breezyweather/sources/ekuk/EkukService.kt

```diff
@@ -0,0 +1,180 @@
+import dagger.hilt.android.qualifiers.ApplicationContext
+import io.reactivex.rxjava3.core.Observable
+import org.breezyweather.common.extensions.code
+import org.breezyweather.common.extensions.currentLocale
+import org.breezyweather.common.source.HttpSource
+import org.breezyweather.common.source.LocationParametersSource
+import org.breezyweather.common.source.WeatherSource
+import org.breezyweather.sources.ekuk.json.EkukObservationsResult
+import retrofit2.Retrofit
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+import java.util.TimeZone
+import javax.inject.Inject
+import javax.inject.Named
+
+class EkukService @Inject constructor(
+    @ApplicationContext context: Context,
+    @Named("JsonClient") client: Retrofit.Builder,
+) : HttpSource(), WeatherSource, LocationParametersSource {
+    override val id = "ekuk"
+    override val name = "EKUK (${Locale(context.currentLocale.code, "EE").displayCountry})"
+    override val continent = SourceContinent.EUROPE
+    override val privacyPolicyUrl = ""
+    override val color = Color.rgb(51, 153, 51)
+    private val weatherAttribution by lazy {
+        with(context.currentLocale.code) {
+            when {
+                startsWith("et") -> "Eesti Keskkonnauuringute Keskus"
+                startsWith("ru") -> "Эстонский центр экологических исследований"
+                startsWith("uk") -> "Естонський центр екологічних досліджень"
+                else -> "Estonian Environmental Research Center"
+            }
+        }
+    }
+
+    private val mApi by lazy {
+        client.baseUrl(EKUK_BASE_URL)
+            .build()
+            .create(EkukApi::class.java)
+    }
+
+    override val supportedFeatures = mapOf(
+        SourceFeature.AIR_QUALITY to weatherAttribution,
+        SourceFeature.POLLEN to weatherAttribution
+    )
+
+    override fun isFeatureSupportedForLocation(
+        location: Location,
+        feature: SourceFeature,
+    ): Boolean {
+        return location.countryCode.equals("EE", ignoreCase = true)
+    }
+
+    override fun requestWeather(
+        context: Context,
+        location: Location,
+        requestedFeatures: List<SourceFeature>,
+    ): Observable<WeatherWrapper> {
+        val airQualityStation = location.parameters.getOrElse(id) { null }?.getOrElse("airQualityStation") { null }
+        val pollenStation = location.parameters.getOrElse(id) { null }?.getOrElse("pollenStation") { null }
+
+        val formatter = SimpleDateFormat("dd.MM.yyyy", Locale.ENGLISH)
+        formatter.timeZone = TimeZone.getTimeZone("Europe/Tallinn")
+        val now = Calendar.getInstance()
+        val today = formatter.format(now.time)
+        now.add(Calendar.DATE, -1)
+        val yesterday = formatter.format(now.time)
+
+        val failedFeatures = mutableListOf<SourceFeature>()
+        val airQuality = if (SourceFeature.AIR_QUALITY in requestedFeatures) {
+            if (!airQualityStation.isNullOrEmpty()) {
```

Comment:
Move `val airQualityStation = location.parameters.getOrElse(id) { null }?.getOrElse("airQualityStation") { null }` above this line

Reply:
Fixed